lib/json-schema/validator.rb

```diff
@@ -55,6 +56,11 @@ def initialize(schema_data, data, opts={})
       @data = initialize_data(data)
       @@mutex.synchronize { build_schemas(@base_schema) }
 
+      # If the :fragment option is set, try and validate against the fragment
+      if opts[:fragment]
+        @base_schema = schema_from_fragment(@base_schema, opts[:fragment])
+      end
```

Comment:
this is moved before the validate_schema block so that the correct schema, at the fragment pointer, is validated